app/controllers/admin/submissions_controller.rb

```diff
@@ -36,12 +48,12 @@ def unflag
 
   def add_tag
     @submission.tag_list.add(admin_submission_params[:tag_list].split(","))
-    @submission.save
+    @submission.save!
```

Comment:
Why the bang here?

Reply:
To raise an exception if there is a problem saving the record.   I ran into an unusual case where I had several submissions and then made a field required on the 2nd page.   When I tagged submissions I had made prior to the required field change, the tags wouldn't be saved.  No error, nothing logged.   I then discovered the submission save was not completed due to the required field validation.   I added the bang so this condition would be raised.   

If this scenario seems likely in the future, we may want to handle that case more gracefully than throwing an exception.  I can write a ticket for that if so.

Reply:
👍